lib/util/utils.js

```diff
@@ -758,3 +758,43 @@ function isPropertyRequired(propName, model) {
  * Contains the reverse mapping of http.STATUS_CODES
  */
 exports.statusCodeStringToStatusCode = lodash.invert(lodash.mapValues(http.STATUS_CODES, value => { return value.replace(/ |-/g, "").toLowerCase(); }));
+
+
+/**
+ * Models an ARM cloud error.
+ */
+exports.CloudError = {
+  description: "Error response describing why the operation failed.",
+  schema: {
```

Comment:
I see what you are trying to do. However if possible do not nest properties. That will make sure that when we use oneOf for nullable types we do not do anything drastic.

Reply:
not sure what you mean. Also here would prefer to not have nullable types since there is no reason for that.